src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,188 @@
+  if first(DBInterface.execute(
+    db,
+    """
+  SELECT COUNT(*) AS n FROM sqlite_master WHERE 
+  name IN ('gpkg_spatial_ref_sys', 'gpkg_contents') AND 
+  type IN ('table', 'view');
+    """
+  )).n != 2
+    throw(ErrorException("missing required metadata tables in the GeoPackage SQL database"))
+  end
+
+  db
+end
+
+# According to Geometry Columns Table Requirements
+# https://www.geopackage.org/spec/#:~:text=2.1.5.1.2.%20Table%20Data%20Values
+#------------------------------------------------------------------------------
+# Requirement 22: gpkg_geometry_columns table
+# SHALL contain one row record for the geometry column
+# in each vector feature data table
+#
+# Requirement 23: gpkg_geometry_columns table_name column
+# SHALL reference values in the gpkg_contents table_name column
+# for rows with a data_type of 'features'
+#
+# Requirement 24: The column_name column value in a gpkg_geometry_columns row
+# SHALL be the name of a column in the table or view specified by the table_name
+# column value for that row.
+#
+# Requirement 25: The geometry_type_name value in a gpkg_geometry_columns row
+# SHALL be one of the uppercase geometry type names specified
+
+# Requirement 26: The srs_id value in a gpkg_geometry_columns table row
+# SHALL be an srs_id column value from the gpkg_spatial_ref_sys table.
+#
+# Requirement 27: The z value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 28: The m value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 146: The srs_id value in a gpkg_geometry_columns table row
+# SHALL match the srs_id column value from the corresponding row in the
+# gpkg_contents table.
+function gpkgextract(db, ; layer=1)
+  rowtable = DBInterface.execute(
+    # According to https://www.geopackage.org/spec/#r16 
+    # Values of the gpkg_contents table srs_id column 
+    # SHALL reference values in the gpkg_spatial_ref_sys table srs_id column
+    # According to https://www.geopackage.org/spec/#r18
+    # The gpkg_contents table SHALL contain a row 
+    # with a lowercase data_type column value of "features" 
+    # for each vector features user data table or view.
+    db,
+    """
+    SELECT g.table_name AS tablename, g.column_name AS geomcolumn, 
+    c.srs_id AS srsid, g.z, srs.organization AS org, srs.organization_coordsys_id AS orgcoordsysid,
+    ( SELECT type FROM sqlite_master WHERE lower(name) = lower(c.table_name) AND type IN ('table', 'view')) AS object_type
+    FROM gpkg_geometry_columns g, gpkg_spatial_ref_sys srs
+    JOIN gpkg_contents c ON ( g.table_name = c.table_name )
+    WHERE c.data_type = 'features'
+    AND object_type IS NOT NULL
+    AND g.srs_id = srs.srs_id
+    AND g.srs_id = c.srs_id
+    AND g.z IN (0, 1, 2)
+    AND g.m = 0
+    LIMIT 1 OFFSET ($layer-1);
+    """
+  )
+
+  # Note: first feature table that is read specifies the CRS to be used on all feature tables resulted from SELECT statement
```

Comment:
I am concerned about this. Shouldn't we load the CRS information from the feature table specified with the `layer` option? Are you positive that the spec forces the same CRS across feature tables?